qkernel/src/kernel_def.rs

```diff
@@ -208,24 +209,54 @@ pub fn Invlpg(addr: u64) {
     }
 }
 
+
+
+
 #[cfg(target_arch = "x86_64")]
 #[inline(always)]
 pub fn HyperCall64(type_: u16, para1: u64, para2: u64, para3: u64, para4: u64) {
+    let vcpuid = GetVcpuId();
+    let share_para_page  = MemoryDef::HYPERCALL_PARA_PAGE_OFFSET as *mut ShareParaPage;
+    let mut share_para =unsafe{&mut (*share_para_page).SharePara[vcpuid]};
+    share_para.para1 = para1;
+    share_para.para2 = para2;
+    share_para.para3 = para3;
+    share_para.para4 = para4;
+
     unsafe {
         let data: u8 = 0;
         asm!("
             out dx, al
             ",
             in("dx") type_,
             in("al") data,
-            in("rsi") para1,
-            in("rcx") para2,
-            in("rdi") para3,
-            in("r10") para4
         )
     }
 }
 
+//VCPU 0 did not set gs as corresponding Vcpulocal address, thus use another function to use default position
+#[cfg(target_arch = "x86_64")]
```

Comment:
why don't you setup the VCPU0's number after vcpu0 enters the guest? In this case you only need one `HyperCall64` function

Reply:
> why don't you setup the VCPU0's number after vcpu0 enters the guest? In this case you only need one `HyperCall64` function

Swapgs to set the vcpu number rely on the sharedspace, we still need a hypercall to initialize the sharedspace struct.